usaspending_api/agency/v2/views/bureau_federal_account.py

```diff
@@ -0,0 +1,105 @@
+    been submitted in File B.
+    """
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/agency/toptier_code/sub_components/bureau_slug.md"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.params_to_validate = ["fiscal_year", "agency_type"]
+
+    @property
+    def bureau_slug(self):
+        return self.kwargs["bureau_slug"]
+
+    @cache_response()
+    def get(self, request: Request, *args: Any, **kwargs: Any) -> Response:
+        self.sortable_columns = ["name", "id", "total_obligations", "total_outlays", "total_budgetary_resources"]
+        self.default_sort_column = "total_obligations"
+        results = sorted(
+            self.get_federal_account_list(),
+            key=lambda x: x.get(self.pagination.sort_key),
+            reverse=self.pagination.sort_order == "desc",
+        )
+        page_metadata = get_pagination_metadata(len(results), self.pagination.limit, self.pagination.page)
+        totals = self.get_totals(results)
+        return Response(
+            {
+                "toptier_code": self.toptier_code,
+                "bureau_slug": self.bureau_slug,
+                "fiscal_year": self.fiscal_year,
+                "totals": totals,
+                "page_metadata": page_metadata,
+                "results": results[self.pagination.lower_limit : self.pagination.upper_limit],
+                "messages": self.standard_response_messages,
+            }
+        )
+
+    def get_totals(self, results) -> dict:
+        totals = {
+            "total_obligations": sum(x["total_obligations"] for x in results),
+            "total_outlays": sum(x["total_outlays"] for x in results),
+            "total_budgetary_resources": sum(x["total_budgetary_resources"] for x in results),
+        }
+        return totals
+
+    def get_federal_account_list(self) -> List[dict]:
+        federal_accounts = [
+            x["federal_account_code"]
+            for x in BureauTitleLookup.objects.filter(bureau_slug=self.bureau_slug).values("federal_account_code")
+        ]
+        filters = [
+            Q(
+                treasury_account_identifier__federal_account__parent_toptier_agency_id=self.toptier_agency.toptier_agency_id
+            ),
+            Q(treasury_account_identifier__federal_account__federal_account_code__in=federal_accounts),
+            Q(fiscal_year=self.fiscal_year),
+            Q(fiscal_period=self.fiscal_period),
+        ]
+
+        last_period_results = (
+            GTASSF133Balances.objects.filter(*filters)
+            .annotate(
+                name=F("treasury_account_identifier__federal_account__account_title"),
+                account_code=F("treasury_account_identifier__federal_account__federal_account_code"),
+            )
+            .values("name", "account_code")
+            .annotate(
+                total_budgetary_resources=Sum("total_budgetary_resources_cpe"),
```

Comment:
Should these be calculated the same as other GTAS sourced values on the COVID page?